pkg/volume/flexvolume/plugin.go

```diff
@@ -64,44 +65,29 @@ func NewFlexVolumePlugin(pluginDir, name string) (volume.VolumePlugin, error) {
 		unsupportedCommands: []string{},
 	}
 
-	// Check whether the plugin is attachable.
-	ok, err := isAttachable(flexPlugin)
+	// Retrieve driver reported capabilities
+	call := flexPlugin.NewDriverCall(initCmd)
+	ds, err := call.Run()
 	if err != nil {
 		return nil, err
 	}
 
-	if ok {
-		// Plugin supports attach/detach, so return flexVolumeAttachablePlugin
+	driverCaps := ds.getDriverCapabilities()
+	flexPlugin.capabilities = driverCaps
+
+	// Check whether the plugin is attachable.
+	if driverCaps.attach {
+		// Plugin supports attach/detach by default, so return flexVolumeAttachablePlugin
 		return &flexVolumeAttachablePlugin{flexVolumePlugin: flexPlugin}, nil
 	} else {
 		return flexPlugin, nil
 	}
 }
 
-func isAttachable(plugin *flexVolumePlugin) (bool, error) {
-	call := plugin.NewDriverCall(initCmd)
-	res, err := call.Run()
-	if err != nil {
-		return false, err
-	}
-
-	// By default all plugins are attachable, unless they report otherwise.
-	cap, ok := res.Capabilities[attachCapability]
-	if ok {
-		// cap is false, so plugin does not support attach/detach calls.
-		return cap, nil
-	}
-
-	return true, nil
-}
-
 // Init is part of the volume.VolumePlugin interface.
 func (plugin *flexVolumePlugin) Init(host volume.VolumeHost) error {
-	plugin.host = host
```

Comment:
@TerraTech this s the cause of nil pointer dereference, please revert this line.

Reply:
nvm, I opened a pr myself since it's just one line https://github.com/kubernetes/kubernetes/pull/51342

Reply:
Oops, that is a BPB mistake, I didn't mean to delete that line.  :(